File: src/BreakfastProvider.Api/Models/Responses/OrderResponse.cs
namespace BreakfastProvider.Api.Models.Responses;

public record OrderResponse
{
    public Guid OrderId { get; init; }
    public string CustomerName { get; init; } = string.Empty;
    public List<OrderItemResponse> Items { get; init; } = [];
    public int? TableNumber { get; init; }
    public string Status { get; init; } = "Created";
    public DateTime CreatedAt { get; init; }
}

public record OrderItemResponse
{
    public string ItemType { get; init; } = string.Empty;
    public Guid BatchId { get; init; }
    public int Quantity { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/response/OrderResponse.java
package io.lemonlion.breakfast.model.response;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

/** Twin of C# {@code OrderResponse}. */
public record OrderResponse(
        UUID orderId,
        String customerName,
        List<OrderItemResponse> items,
        Integer tableNumber,
        String status,
        Instant createdAt) {
}
